### PR TITLE
feat: allow notarization on macOS versions down to 10.15

### LINKED ISSUE #200: Allow notarization on macOS versions down to 10.15
With `altool` being deprecated and `notarytool` being available from Xcode 13, which requires macOS 11.3, it seems not possible to notarize apps on an older macOS version. Yet Apple provided a hacky way to achieve this, which is taking the `notarytool` binary from a newer os version and copying it to an older one. It works down to 10.15 ([source](https://developer.apple.com/documentation/technotes/tn3147-migrating-to-the-latest-notarization-tool#Enable-notarization-on-an-older-version-of-macOS)).

The most obvious way to do this is to copy that binary into the directory that is normally used to keep `notarytool`, which is `/Applications/Xcode.app/Contents/Developer/usr/bin/notarytool`. Unfortunately, running it with `xcrun` produces following output:
```
pawel@mac ~ % xcrun notarytool --version
xcrun: error: unable execute utility "/Applications/Xcode.app/Contents/Developer/usr/bin/notarytool" because it requires a newer version of macOS.
```
Running `notarytool` directly, omitting `xcrun`, works fine.

Since this project uses `xcrun` to run `notarytool` it does not handle this scenario. My proposition is to add an option called `notarytoolPath` that would contain path to the `notarytool` binary. That binary would be called directly if provided. Proposed changes are in the linked pull request.

For anyone trying to get notarization to work using `notarytool` binary copied from a newer macOS: if you run into errors saying that you are missing `/usr/lib/swift/libswift_Concurrency.dylib`, use a binary extracted from Xcode 13.4.1 or Xcode CL Tools 13.4 and the error will go away.

### PR DESCRIPTION
**Pull request by pjaspinski**
More info in the linked issue: closes #200

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -47,6 +47,10 @@ export interface NotaryToolNotarizeAppOptions {
   appPath: string;
 }
 
+export interface NotaryToolOptions {
+  notarytoolPath?: string;
+}
+
```
Comment (erickzhao) [inline src/types.ts:53]: nit: I think I'd prefer just merging this with the type that accepts `appPath` to avoid having too many single-use interfaces.

**Review by erickzhao — COMMENTED** [review]
@pjaspinski I also had a question since I'm not super familiar with legacy versions of macOS:

Apple's Notarization troubleshooting documentation seems to tell users to use macOS SDK 10.9 or later because Mavericks changed how codesigning works:
https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/resolving_common_notarization_issues#3087723

Do we want to align with that 10.9 minimum in the documentation in this case as well?

**Comment by pjaspinski** [comment]
@erickzhao I made a mistake - `notarytool` is supported down to macOS version 10.15 (not 10.5). So the requirement that you mentioned is always met.

File: README.md [hunk 2]
```diff
@@ -39,12 +39,19 @@ For notarization, you need the following things:
   If you are using Electron 11 or below, you must add the `com.apple.security.cs.allow-unsigned-executable-memory` entitlement too.
   When using version 12+, this entitlement should not be applied as it increases your app's attack surface.
 
+### Notarization on older macOS versions
+
+Xcode 13 is available from macOS 11.3, but notarization can be performed on systems down to macOS 10.5 ([source](https://developer.apple.com/documentation/technotes/tn3147-migrating-to-the-latest-notarization-tool#Enable-notarization-on-an-older-version-of-macOS)).
```
Comment (erickzhao) [inline README.md:44]:
```suggestion
Xcode 13 is available from macOS 11.3, but notarization can be performed on systems down to macOS 10.15
(see [TN3147](https://developer.apple.com/documentation/technotes/tn3147-migrating-to-the-latest-notarization-tool#Enable-notarization-on-an-older-version-of-macOS) for more information).
```

**Review by erickzhao — CHANGES_REQUESTED** [review]
One more doc fix, but the overall concept LGTM

**Review by erickzhao — APPROVED** [review]

File: src/notarytool.ts [hunk 3]
```diff
@@ -12,6 +12,12 @@ import { NotaryToolCredentials, NotaryToolStartOptions } from './types';
 
 const d = debug('electron-notarize:notarytool');
 
+function runNotaryTool(args: string[], notarytoolPath?: string) {
+  const useXcrun = typeof notarytoolPath === 'undefined';
```
Comment (dsanders11) [inline src/notarytool.ts:16]:
```suggestion
  const useXcrun = notarytoolPath === undefined;
```

No need for `typeof` here, `notarytoolPath` is declared as a parameter.

File: src/notarytool.ts [hunk 4]
```diff
@@ -43,17 +49,25 @@ function authorizationArgs(rawOpts: NotaryToolCredentials): string[] {
...
   } catch (e) {
     d('failed to pull notarization logs', e);
   }
 }
 
-export async function isNotaryToolAvailable() {
-  const result = await spawn('xcrun', ['--find', 'notarytool']);
-  return result.code === 0;
+export async function isNotaryToolAvailable(notarytoolPath?: string) {
+  if (typeof notarytoolPath !== 'undefined') {
```
Comment (dsanders11) [inline src/notarytool.ts:64]:
```suggestion
  if (notarytoolPath !== undefined) {
```

No need for `typeof` here, `notarytoolPath` is declared as a parameter.

**Comment by felixrieseberg** [comment]
This is a good PR and we'll merge it in, I just want to be clear: We're _not_ supporting older versions of macOS. Using a custom notarization tool is fully unsupported.

**Pull request merged**